Rakefile

```diff
@@ -0,0 +1,114 @@
+require 'json'
+require 'faraday'
+require 'fileutils'
+require 'logger'
+require 'rubygems'
+
+def logger
+  @logger ||= Logger.new STDOUT
+end
+
+def files
+  @files ||= []
+end
+
+task :default => [:update_static_files]
+
+task 'assets:precompile' => :update_static_files
+
+def version_for(version_str)
+  Gem::Version.new version_str.match(/\d+(\.\d+)*/)[0]
+end
+
+def fetch_githubusercontent_file(from, to = nil)
+  conn = Faraday.new('https://raw.githubusercontent.com')
+  public_files_dir = "public/files"
+  to = File.basename(from) unless to
+  to = File.join("..", public_files_dir, to)
+
+  FileUtils.mkdir_p public_files_dir
+
+  response = conn.get do |req|
+    logger.info "Fetching #{conn.url_prefix.to_s}#{from}"
+    req.url from
+  end
+
+
+  dest = File.expand_path(to, __FILE__)
+
+  if response.success?
+    logger.info "Writing to #{dest}"
+    File.write(dest, response.body)
+    logger.info "Setting mode 'a+rx' on #{dest}"
+    FileUtils.chmod "a+rx", dest
+    files << dest
+  else
+    fail "Could not fetch #{from}"
+  end
+end
+
+
+def latest_release_for(repo)
+  conn = Faraday.new('https://api.github.com')
+
+  response = conn.get do |req|
+    releases_url = "repos/#{repo}/releases"
+    logger.info "Fetching releases from #{conn.url_prefix.to_s}#{releases_url}"
+    req.url releases_url
+    oauth_token = ENV['GITHUB_OAUTH_TOKEN']
+    if oauth_token
+      req.headers['Authorization'] = "token #{oauth_token}"
+    end
+  end
+
+  if response.success?
+    json_data = JSON.parse(response.body)
+    fail "No releases found for #{repo}" if json_data.empty?
+    json_data.sort { |a,b| version_for(a["name"]) <=> version_for(b["name"]) }.last["name"]
+  else
+    fail "Could not find releases for #{repo}"
+  end
+end
+
+desc 'update casher'
+task :casher do
+  fetch_githubusercontent_file 'travis-ci/casher/production/bin/casher'
+end
+
+desc 'update gimme'
+task :gimme do
+  latest_release = latest_release_for 'travis-ci/gimme'
+  logger.info "Latest gimme release is #{latest_release}"
+  fetch_githubusercontent_file "travis-ci/gimme/#{latest_release}/gimme"
+end
+
+desc 'update nvm.sh'
+task :nvm do
+  latest_release = latest_release_for 'creationix/nvm'
+  node_js_rb_path = File.expand_path('../lib/travis/build/script/node_js.rb', __FILE__)
+
+  logger.info "Latest nvm release is #{latest_release}"
+  sed_cmd = %Q(sed -i "s,^\\(.*\\)NVM_VERSION\s*=.*$,\\1NVM_VERSION = '#{latest_release.gsub(/^v/,'')}'," #{node_js_rb_path})
+
+  logger.info "Updating #{node_js_rb_path}"
+  `#{sed_cmd}`
```

Comment:
This munging of `./lib/travis/build/script/node_js.rb` seems out of place to me.  Could it be a separate target, maybe, so that the expectation is that it's performed manually?  I wouldn't want the source code to be changed during heroku deploy.